app/src/main/java/com/example/chatbotandroid/controller/MainActivity.kt

```diff
@@ -0,0 +1,108 @@
+package com.example.chatbotandroid.controller
+
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import com.example.chatbotandroid.R
+import androidx.recyclerview.widget.LinearLayoutManager
+import com.example.chatbotandroid.model.ChatAdapter
+import com.example.chatbotandroid.model.Message
+import com.example.chatbotandroid.utils.Constants
+import com.example.chatbotandroid.utils.Time
+import org.java_websocket.client.WebSocketClient
+import kotlinx.android.synthetic.main.activity_main.*
+import kotlinx.coroutines.*
+import org.java_websocket.handshake.ServerHandshake
+import java.lang.Exception
+import java.net.URI
+
+class MainActivity : AppCompatActivity() {
+    private lateinit var adapter: ChatAdapter
+    private lateinit var webSocketClient: WebSocketClient
+    private val webSocketUri: URI? = URI(Constants.WEB_SOCKET_URL)
+
+    //Life-cycle methods
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_main)
+        recyclerView()
+        registerOnClickListeners()
+    }
+
+    override fun onResume() {
+        super.onResume()
+        initWebSocket()
+    }
+
+    override fun onPause() {
+        super.onPause()
+        //Close web socket if app is on background to avoid unnecessary active connection
+        webSocketClient.close()
+    }
+
+    //Websocket methods
+    private fun initWebSocket() {
+        createWebSocketClient(webSocketUri)
+        webSocketClient.connect()
+    }
+
+    private fun createWebSocketClient(wsURI: URI?) {
+        webSocketClient = object : WebSocketClient(wsURI) {
+            override fun onOpen(handshakedata: ServerHandshake?) {
+                println("Websocket connection open")
+            }
+            override fun onMessage(message: String?) {
+                if (message != null) {
+                    appendServerResponse(message)
+                }
+            }
+            override fun onClose(code: Int, reason: String?, remote: Boolean) {
+                println(reason)
```

Comment:
there's a convenient android logger that can log messages with a tag, making it easy to see them in android studio's logcat with other junk filtered out. it's the Log class, e.g. `Log.d("my_tag", reason)`